berkeley-mobile/StudyPact/Profile/ProfileViewController.swift

```diff
@@ -445,3 +460,40 @@ extension ProfileViewController {
         dismiss(animated: true, completion: nil)
     }
 }
+
+extension ProfileViewController {
+    func validateFacebook(text: String) -> String {
+        if CharacterSet.decimalDigits.isSuperset(of: CharacterSet(charactersIn: text)) {
+            return text
+        }
+        
+        let fbPattern = ##"(?:(?:http|https):\/\/)?(?:www.)?facebook.com\/(?:(?:\w)*#!\/)?(?:pages\/)?(?:[?\w\-]*\/)?(?:profile.php\?id=(?=\d.*))?([\w\-]*)?"##
```

Comment:
we only want their username so the facebook.com wouldn't be included

Reply:
The regex checks for either the url with facebook.com/123456789 or just 123456789, and the result is just the ID